### PR TITLE
VACMS-4520 4523 Disable title and display intro

### PR DESCRIPTION
**Pull request by ethanteague**
## Description
See #4520 #4523

## Testing done
Visual / Behat

## Screenshots
image
image

## QA steps
- As an administrator, go to `node/2360/edit` and visually verify that you can alter the title.
- Logout, login back in as a user without administrator role, and visually verify that the title field is no longer editable.
- As an administrator, go to `/health-care/covid-19-vaccine-tag` and visually verify that the page introduction is visible.

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Core Application Team`
- [x] `Product Support Team`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change? 
- [ ] Yes, and it's written in issue ____ and queued for publication. 
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written 
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change. 
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module [hunk 1]
```diff
@@ -423,6 +424,13 @@ function _va_gov_toggle_field_states(array &$form, array $toggle_group) {
  */
 function va_gov_backend_form_alter(&$form, FormStateInterface $form_state, $form_id) {
 
+  // Prevent system title from being edited on the edit form by non admins.
```
Comment (kevwalsh) [inline docroot/modules/custom/va_gov_backend/va_gov_backend.module:427]: oh nice, i like this approach, it takes care of #4522 as well. Can you extend this to content_admin, not just admin?

**Review by kevwalsh — DISMISSED** [review]

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module (hunk 1 above)
Comment (ethanteague) [reply docroot/modules/custom/va_gov_backend/va_gov_backend.module:427]: Nice thinking, @kevwalsh ! Updated :+1:

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module [hunk 2]
```diff
@@ -423,6 +424,19 @@ function _va_gov_toggle_field_states(array &$form, array $toggle_group) {
...
+    $allowed_roles = [
+      'administrator',
+      'content_admin',
+    ];
+    $current_user_roles = User::load(\Drupal::currentUser()->id())->getRoles();
+    $admin_roles = count(array_intersect($allowed_roles, $current_user_roles));
+    if ($admin_roles < 1) {
+      $form['title']['#disabled'] = 'disabled';
+    }
+  }
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_backend/va_gov_backend.module:438]: Can we chunk this into a meaningfully named local function and call it just to keep all the logic bloat out of the hook_form_alter()?

**Review by swirtSJW — CHANGES_REQUESTED** [review]

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module (hunk 2 above)
Comment (ethanteague) [reply docroot/modules/custom/va_gov_backend/va_gov_backend.module]: Yep, @swirtSJW :+1: 

We should probably clean this big 'ol nasty alter up one of these days. It's grown to where it's no longer readable.

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module [hunk 3]
```diff
@@ -698,6 +704,24 @@ function _va_gov_backend_get_unlocked_alias_bundles() {
...
+ *
+ * @param array $form
+ *   The node form array.
+ */
+function _va_gov_backend_vamc_title_access(array &$form) {
+  $allowed_roles = [
+    'administrator',
+    'content_admin',
+  ];
+  $current_user_roles = User::load(\Drupal::currentUser()->id())->getRoles();
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_backend/va_gov_backend.module:718]: Sorry, Missed this earlier.  This seems redundant.   User::load   and \Drupal::currentUser()   both give you the same [account user interface](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Session%21AccountProxyInterface.php/interface/AccountProxyInterface/8.7.x)  
which provides you with  the getRoles() method 
I think you could get away with 
`$current_user_roles = \Drupal::currentUser()->getRoles();`
and avoid the redundant load of a user you already have.

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module [hunk 4]
```diff
@@ -698,6 +704,24 @@ function _va_gov_backend_get_unlocked_alias_bundles() {
...
+  $admin_roles = count(array_intersect($allowed_roles, $current_user_roles));
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_backend/va_gov_backend.module:719]: Bonus points for use of array intersect.  `$admin_roles` is kind of a misrepresentation though... it does not contain roles, it is a count  so `$allowed_role_count` would be more accurate.

**Review by swirtSJW — CHANGES_REQUESTED** [review]

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module (hunk 3 above)
Comment (ethanteague) [reply docroot/modules/custom/va_gov_backend/va_gov_backend.module]: Argh - I knew that, but still didn't catch it. Apologies, @swirtSJW - ty for the keen eyes :+1:

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module (hunk 4 above)
Comment (ethanteague) [reply docroot/modules/custom/va_gov_backend/va_gov_backend.module]: Agreed, @swirtSJW :+1: - updated

**Review by swirtSJW — APPROVED** [review]
Nice work Ethan.

**Comment by swirtSJW** [comment]
image

**Pull request merged**